src/ci_workflow/ci_check_manifest_component.py

```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+
+from ci_workflow.ci_check import CiCheckDist
+from manifests.build_manifest import BuildManifest
+
+
+class CiCheckManifestComponent(CiCheckDist):
+    class MissingComponentError(Exception):
+        def __init__(self, component, url):
+            super().__init__(f"Missing {component} in {url}.")
+
+    def check(self):
+        for architecture in ["x64", "arm64"]:
```

Comment:
Can we import from build args where this would be verified rather than duplicate? https://github.com/opensearch-project/opensearch-build/blob/fba6709809ebe0e51b6f856e0ebeaa1de52e655e/src/build_workflow/build_args.py#L14-L17